community/modules/scripts/spack-install/variables.tf

```diff
@@ -177,12 +189,41 @@ EOT
 }
 
 variable "environments" {
-  description = "Defines a spack environment to configure."
-  default     = null
-  type = list(object({
-    name     = string
-    packages = list(string)
-  }))
+  description = <<EOT
+  Defines spack environments to configure, given as a list.
+  Each environment must define a name, and type.
+  Valid types for environments are 'file', and 'packages'.
+  Environments of type 'packages' must define a list named packages, containing
+  all of the packages to install in that environment.
+  Environments of type 'file' must define a string named 'content' containing the
+  contents of a spack.yaml file to create the environment from.
```

Comment:
nit: although it ends up in `spack.yaml` I would rather say something like "contents of Spack YAML environment file" which adopts the headers you'd find in their Read The Docs